### PR TITLE
HDDS-8951. Audit log from SCMClientProtocolServer has no information about request params

### PR DESCRIPTION
**Pull request by Tejaskriya**
## What changes were proposed in this pull request?

Many audit log messages in SCMClientProtocolServer did not include any request parameters and passed null to buildAuditMessage*(). Audit maps have been created where necessary and have been passed to buildAuditMessage*() calls.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8951

## How was this patch tested?

Tested manually

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java [hunk 1]
```diff
@@ -683,15 +683,25 @@ public void closeContainer(long containerID) throws IOException {
...
       throws IOException {
+    Map<String, String> auditMap = Maps.newHashMap();
+    if (type != null) {
+      auditMap.put("replicationType", type.toString());
+    }
+    if (factor != null) {
+      auditMap.put("replicationFactor", factor.toString());
+    }
+    if (nodePool != null) {
+      auditMap.put("nodePool", nodePool.getNodesList().toString());
```
[the remark this answers is not in the document]
Comment (Tejaskriya) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: I have fixed it now

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java [hunk 2]
```diff
@@ -683,15 +683,29 @@ public void closeContainer(long containerID) throws IOException {
...
       HddsProtos.ReplicationFactor factor, HddsProtos.NodePool nodePool)
       throws IOException {
+    Map<String, String> auditMap = Maps.newHashMap();
+    if (type != null) {
+      auditMap.put("replicationType", type.toString());
+    }
+    if (factor != null) {
+      auditMap.put("replicationFactor", factor.toString());
+    }
+    if (nodePool != null) {
```
Comment (sadanand48) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:693]:
```suggestion
    if (nodePool != null && !nodePool.getNodesList().isEmpty()) {
```
maybe also add an empty check?

**Review by sadanand48 — APPROVED** [review]
Thanks @Tejaskriya for addressing the comments, the changes look good to me.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java [hunk 3]
```diff
@@ -683,15 +683,29 @@ public void closeContainer(long containerID) throws IOException {
   public Pipeline createReplicationPipeline(HddsProtos.ReplicationType type,
       HddsProtos.ReplicationFactor factor, HddsProtos.NodePool nodePool)
       throws IOException {
+    Map<String, String> auditMap = Maps.newHashMap();
```
Comment (aryangupta1998) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:686]: I see we have used the final keyword for the map variable except here but I would suggest we can remove the final keyword for the map variable in other places as it won't have any effect, if we really want the map values shouldn’t change once initialized we should use `Collections.unmodifiableMap(map)` but I feel we don't need it here, so I would suggest we can remove the final keyword for the map variables in this PR.

Comment (Tejaskriya) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:686]: Thank you for the review and suggestion! I have removed the final keyword form auditMaps now.

**Review by aryangupta1998 — APPROVED** [review]
Thanks, @Tejaskriya for the PR and @sadanand48 for the review!

**Pull request merged**